### PR TITLE
Add testing of Parquet files from apache/parquet-testing [databricks]

### PR DESCRIPTION
**Pull request by jlowe**
This adds apache/parquet-testing as a git submodule to leverage as Parquet read tests during integration tests.  The files are added under src/test/resources in the integration_tests tarball that is built for distributed filesystem testing and will be found via the std_input_path searching. If not found there, it will attempt to find the files in a relative location to the integration test file.  Failure to find the input files is an error for a precommit or nightly test build.

### CODE REVIEW

**Comment by jlowe** [comment]
build

**Review by tgravescs — DISMISSED** [review]

**Comment by tgravescs** [comment]
ci had failed not able to find parquet_testing, is it just a submodule type issue?

File: integration_tests/src/main/python/parquet_testing_test.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    if std_path: places.append(Path(std_path))
+    places.append(Path(__file__).parent.joinpath("../../../../thirdparty").resolve())
+    for p in places:
+        files = []
+        for pattern in glob_patterns:
+            files += p.glob(pattern)
+        if files:
+            return files
+    if is_at_least_precommit_run():
+        raise AssertionError("Cannot find parquet_testing data in any of: " +
```
Comment (tgravescs) [inline integration_tests/src/main/python/parquet_testing_test.py:86]: nit: parquet-testing is the name of the thirdparty dir

**Comment by jlowe** [comment]
> ci had failed not able to find parquet_testing, is it just a submodule type issue?

Yes, need to update CI to update/init submodules when pulling from spark-rapids repo.

**Comment by pxLi** [comment]
we also need to update build/test scripts like
https://github.com/NVIDIA/spark-rapids/blob/branch-23.08/build/buildall
and scripts under https://github.com/NVIDIA/spark-rapids/tree/branch-23.08/jenkins


```
The files are added under src/test/resources in the integration_tests tarball that is built for distributed filesystem testing and will be found via the std_input_path searching. If not found there, it will attempt to find the files in a relative location to the integration test file. Failure to find the input files is an error for a precommit or nightly test build.
```
there should be some risks to block nightly CIs (too many scenarios to cover w/ the new submodule init), 
can we first make it configurable by adding an arg and default as disabled so that we could enable tests in different CI ENVs one by one? like enable pre-merge for now thanks

**Comment by jlowe** [comment]
> can we first make it configurable by adding an arg and default as disabled so that we could enable tests in different CI ENVs one by one?

I updated the PR to add a `--force_parquet_testing_tests` cmdline option that is disabled by default, but when enabled will force the parquet-testing tests to fail tests if it cannot find the input data.  This will allow each nightly script to enable testing by adding this flag.

I also updated the PR so it will fail parquet-testing by default in pre-commit builds if the data is missing but not in any other build types, including nightly.

**Comment by jlowe** [comment]
@pxLi looks like the most recent 322 compile workflow somehow ran with JDK11 (or something else not JDK8) and failed but the others did not.  Any idea how that could happen?
```
Warning:  [Warn] : bootstrap class path not set in conjunction with -source 8
Error: ] : warnings found and -Werror specified
```

**Comment by pxLi** [comment]
> @pxLi looks like the most recent 322 compile workflow somehow ran with JDK11 (or something else not JDK8) and failed but the others did not. Any idea how that could happen?
> 
> ```
> Warning:  [Warn] : bootstrap class path not set in conjunction with -source 8
> Error: ] : warnings found and -Werror specified
> ```

Saw that in github actions CI, seems weird as the java runtime is only setup as 8,
let me take a look later

**Comment by NvTimLiu** [comment]
I built the PR locally, and got a RAT check failure as below cc @jlowe 

```
  B     thirdparty/parquet-testing/data/rle-dict-uncompressed-corrupt-checksum.parquet
 !????? thirdparty/parquet-testing/data/_KEY_MATERIAL_FOR_external_key_material_java.parquet.encrypted.json
  B     thirdparty/parquet-testing/data/alltypes_tiny_pages.parquet
```

**Comment by razajafri** [comment]
fixes #8708

**Comment by NvTimLiu** [comment]
> fixes #8708

Oh I got below parquet_testing_test errors with the latest commit

```
12:28:26  =========================== short test summary info ============================
12:28:26  ERROR ../../src/main/python/parquet_testing_test.py - AssertionError: Cannot find parquet-testing data in any of: /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/integration_tests/src/test/resources/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/integration_tests/src/test/resources/parquet-testing/bad_data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/thirdparty/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/thirdparty/parquet-testing/bad_data/*.parquet
12:28:26  ERROR ../../src/main/python/parquet_testing_test.py - AssertionError: Cannot find parquet-testing data in any of: /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/integration_tests/src/test/resources/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/integration_tests/src/test/resources/parquet-testing/bad_data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/thirdparty/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/thirdparty/parquet-testing/bad_data/*.parquet
12:28:26  ERROR ../../src/main/python/parquet_testing_test.py - AssertionError: Cannot find parquet-testing data in any of: /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/integration_tests/src/test/resources/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/integration_tests/src/test/resources/parquet-testing/bad_data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/thirdparty/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/thirdparty/parquet-testing/bad_data/*.parquet
12:28:26  ERROR ../../src/main/python/parquet_testing_test.py - AssertionError: Cannot find parquet-testing data in any of: /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/integration_tests/src/test/resources/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/integration_tests/src/test/resources/parquet-testing/bad_data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/thirdparty/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/thirdparty/parquet-testing/bad_data/*.parquet
12:28:26  ERROR ../../src/main/python/parquet_testing_test.py - AssertionError: Cannot find parquet-testing data in any of: /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/integration_tests/src/test/resources/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/integration_tests/src/test/resources/parquet-testing/bad_data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/thirdparty/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7521-ci-2/thirdparty/parquet-testing/bad_data/*.parquet
12:28:26  = 17875 passed, 1359 skipped, 633 xfailed, 276 xpassed, 69 warnings, 5 errors in 4545.82s (1:15:45) =
```

**Comment by jlowe** [comment]
> got a RAT check failure as below

Nice catch!  That should be caught by the git workflows once they're updated to do a submodule update before trying to run `mvn verify`.  Wondering if that should be handled by #8684.

I'll update for the RAT check and test failures shortly.

**Review by revans2 — APPROVED** [review]

**Review by NvTimLiu — APPROVED** [review]
LGTM, test PASS with premerge CI and  DB integration too.

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
hmm, weird failure to checkout merged commit in databricks stage. retriggered to see if its a glitch of github
```
02:30:01  [2023-07-20T18:29:59.357Z]  > git fetch --tags --force --progress -- https://github.com/nvidia/spark-rapids +refs/pull/*/merge:refs/remotes/origin/pr/* # timeout=10
02:30:01  [2023-07-20T18:30:01.289Z]  > git rev-parse 15cff70bb1c14e23c762cc4fed716a759916220d^{commit} # timeout=10
02:30:01  [2023-07-20T18:30:01.413Z]  > git rev-parse origin/15cff70bb1c14e23c762cc4fed716a759916220d^{commit} # timeout=10
02:30:01  [2023-07-20T18:30:01.482Z]  > git rev-parse 15cff70bb1c14e23c762cc4fed716a759916220d^{commit} # timeout=10
02:30:01  [2023-07-20T18:30:01.543Z] ERROR: Couldn't find any revision to build. Verify the repository and branch configuration for this job.
```

**Comment by revans2** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
currently CI will fail
```
[2023-07-24T02:05:54.517Z] =========================== short test summary info ============================

[2023-07-24T02:05:54.517Z] ERROR ../../src/main/python/parquet_testing_test.py - AssertionError: Cannot find parquet-testing data in any of: /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/integration_tests/src/test/resources/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/integration_tests/src/test/resources/parquet-testing/bad_data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/thirdparty/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/thirdparty/parquet-testing/bad_data/*.parquet

[2023-07-24T02:05:54.517Z] ERROR ../../src/main/python/parquet_testing_test.py - AssertionError: Cannot find parquet-testing data in any of: /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/integration_tests/src/test/resources/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/integration_tests/src/test/resources/parquet-testing/bad_data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/thirdparty/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/thirdparty/parquet-testing/bad_data/*.parquet

[2023-07-24T02:05:54.517Z] ERROR ../../src/main/python/parquet_testing_test.py - AssertionError: Cannot find parquet-testing data in any of: /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/integration_tests/src/test/resources/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/integration_tests/src/test/resources/parquet-testing/bad_data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/thirdparty/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/thirdparty/parquet-testing/bad_data/*.parquet

[2023-07-24T02:05:54.517Z] ERROR ../../src/main/python/parquet_testing_test.py - AssertionError: Cannot find parquet-testing data in any of: /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/integration_tests/src/test/resources/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/integration_tests/src/test/resources/parquet-testing/bad_data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/thirdparty/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/thirdparty/parquet-testing/bad_data/*.parquet

[2023-07-24T02:05:54.517Z] ERROR ../../src/main/python/parquet_testing_test.py - AssertionError: Cannot find parquet-testing data in any of: /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/integration_tests/src/test/resources/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/integration_tests/src/test/resources/parquet-testing/bad_data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/thirdparty/parquet-testing/data/*.parquet, /home/jenkins/agent/workspace/jenkins-rapids_premerge-github-7592-ci-2/thirdparty/parquet-testing/bad_data/*.parquet
```

as we forced the assert in premerge
`if is_precommit_run() or is_parquet_testing_tests_forced():`

will retry after merging https://github.com/NVIDIA/spark-rapids/pull/8684

**Comment by NvTimLiu** [comment]
We should merge https://github.com/NVIDIA/spark-rapids/pull/8684 into this PR to git clone the parquet-testing submodule for the pre-merge CI, otherwise the pre-merge CI will be failed due to `Cannot find parquet-testing data in any of : integration_tests/src/test/resources/parquet-testing/data/*.parquet`

**Comment by pxLi** [comment]
> We should merge #8684 into this PR to git clone the parquet-testing submodule for the pre-merge CI, otherwise the pre-merge CI will fail on `Cannot find parquet-testing data in any of : integration_tests/src/test/resources/parquet-testing/data/*.parquet`

merged #8646. going to retrigger the CI

**Comment by pxLi** [comment]
build

**Pull request merged**